### PR TITLE
Update issue templates into 3 sections

### PR DESCRIPTION
**Pull request by choldgraf**
I find it kind-of awkward to combine our "background and proposal" topics into a single section. It feels a bit wordy, and tends to result in intermingling the two.

This PR proposes separating them into two sections: one for "Context" and one for "Proposal". The goal here is to make the proposal really explicit and concise, and separate it from the "background information".

### CODE REVIEW

**Comment by yuvipanda** [comment]
I already find the extra set of textboxes intimidating. I have sometimes resorted to either opening blank issues (and felt guilty that maybe I'm not providing information that is expected of me to others) and a couple times opened issue but then got stuck at 'hmm, what do i put here?' and then it gets never filed. With this, there'll be 4 textboxes (I think?), and opening an issue becomes a fairly high energy investment task.

**Comment by choldgraf** [comment]
So my goal with this issue is to try and make it clear what kind of text goes in each section, in the hopes that this makes it a bit simpler to wrap one's head around. To me, it is simpler to have 4 major sections, each with a specific role:

```
# Context
# Proposal
# Implementation guide
# Updates and actions
```

Rather than 3 sections with multiple meanings:

```
# Background and proposal
# Implementation guide and constraints
# Updates and actions
```

If the complexity / volume of the text is an issue for folks, we could also explore removing all of the pre-filled text in general. I think having tightly-scoped sections makes it more apparent what should go in each.

Can you go into more detail about where the complexity starts to impede your workflow around the issue forms? Is it having too many words in general == overwhelming? Is it feeling like you have to answer 4 questions instead of one?

**Comment by yuvipanda** [comment]
> Is it having too many words in general == overwhelming? Is it feeling like you have to answer 4 questions instead of one?

Pretty much this!

I also think that it feels somehow non-iterative. I think issues generally tend to develop over time - it's alright to open an issue with a one-line description that gets edited over time, rather than have it all be up-front. The current structure feels like it forces all this thinking upfront, which is valuble too - but comes at a cognitive cost.

**Comment by choldgraf** [comment]
Hmmm, OK let me try and push another commit that cuts down on the number of words, and makes everything after `context` optional, and see how this feels.

**Comment by choldgraf** [comment]
@yuvipanda what do you (or others) think about the latest commit. This significantly reduces the wordiness etc of the page. Here's a screenshot for example of what the issue template would look like:

image

**Comment by yuvipanda** [comment]
@choldgraf definitely an improvement! What is the difference between 'Proposal' and 'Implementation guide'?

**Comment by choldgraf** [comment]
In my mind:

Proposal = the end state you want to achieve.

Implementation guide = anything that helps a person actually get to that end state. Eg "references for inspiration, things to explicitly avoid, etc".

They don't both always have to exist, sometimes the implementation is obvious enough from the proposal, but in that case we just don't need to fill out that section.

**Comment by yuvipanda** [comment]
@choldgraf IMO, I think those two should be combined. So you have one box for the problem to be solved and one for how you want to solve it. i think having sections we don't expect to be filled out is the core of the problem for me, as it requires quite a bit of effort to figure out 'do i need to fill this out?'.

**Comment by choldgraf** [comment]
What do you think about the latest commit? Image below:

image

Anybody else on the @2i2c-org/tech-team have strong opinions? (maybe @damianavila can weigh in with his project management hat on :-) )

**Comment by yuvipanda** [comment]
I like it :) I think I also want to automate the last textbox with a github action - so I can say in a comment 'Action Item: <something>' and it'll automatically copy that into the last section, keep it up todate. I don't think 'progress' gets filled in when you create the issue, but that's something we can deal with later.

Thank you for taking this under consideration, @choldgraf!

**Review by sgibson91 — APPROVED** [review]
I like it - but why do we need both a YAML file and a Markdown file?

**Review by GeorgianaElena — APPROVED** [review]
I like it too 🚀 and also share @sgibson91's curiosity around the two config files. Are they required to render this new template form format?

**Review by damianavila — APPROVED** [review]
LGTM, I like the simplification!
And I think it is a good compromise between having something agile (if you need to drop something to be specified later down the road) and something structure (in case you want to come up with a "full" issue from the very beginning).

**Comment by choldgraf** [comment]
Woohoo! Ok I'll merge. The reason that we have a markdown file is because of the meta repository. It is private, and I don't believe that issue forms work yet on private repositories.

When this discussion is resolved, we should be able to delete the markdown version:

- https://github.com/github/feedback/discussions/4264

File: .github/ISSUE_TEMPLATE/01_new-issue.yml [hunk 1]
```diff
@@ -2,55 +2,30 @@ name: 💡 General Issue
...
-      value: |
-        **Context**
-        Background information: When ___ kind of person is trying to do ___, then ___ happens.
-        Problem or opportunity:  This is a problem because ___. It would be better if ___.
-
-        **Proposed solution**
-        Action to take: We should make it possible to ___ by doing ___.
-        Value and who benefits: This would allow ___ users to ___.
-
+        Any background information that helps others take understand this issue and why it is important.
```
Comment (choldgraf) [inline .github/ISSUE_TEMPLATE/01_new-issue.yml]:
```suggestion
        Any background information that helps others understand this issue and why it is important.

```

Comment (damianavila) [reply .github/ISSUE_TEMPLATE/01_new-issue.yml]: I think you forgot to also update this one in the md file, maybe?
